### DISCUSSION TITLE
[v5] ManyToMany Seeding

### DISCUSSION THREAD

**Discussion by bnason** [adonisjs/core#2181] [Help] [answered]
Is it possible to seed a many to many relationship?

I have a ManyToMany relationship between Project and User and want to have users 'added' to the project during seeding. If it is possible, is it then possible to re-use users so they aren't all unique and one user may belong to multiple projects?

**Comment by wfelipe99** [comment, marked as answer]
Yes, it's possible. What have you tried? It's like a normal seeder, nothing need to change. You'll need to make a model (probably it won't be used in your code, it's just to populate your relationship) so you can use .createMany() like as usual.

The created model might look like this:

```typescript
import { DateTime } from 'luxon'
import { BaseModel, column } from '@ioc:Adonis/Lucid/Orm'

export default class UsersProjects extends BaseModel {
  public static table = 'users_projects'

  @column({ isPrimary: true })
  public id: number

  @column()
  public userId: number

  @column()
  public projectId: number

  @column.dateTime({ autoCreate: true })
  public createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  public updatedAt: DateTime
}
```

Then you can seed it this way:

```typescript
import BaseSeeder from '@ioc:Adonis/Lucid/Seeder'
import UsersProjects from 'App/Models/UsersProjects'

export default class UsersProjectsSeeder extends BaseSeeder {
  public async run() {
    await UsersProjects.createMany([
      {
        userId: 1,
        projectId: 1,
      },
      {
        userId: 1,
        projectId: 2,
      },
      {
        userId: 2,
        projectId: 1,
      },
    ])
  }
}
```

**Comment by willonf** [comment]
I'm with the same doubt and reading the last answer I could understand how to solve the problem. But another doubt comes with this answer. The Adonis documentation say there is no need to create a manyToMany model. So, how can I populate my pivot table on database using, e.g, postman or insomnia? Looking the problem, it seems that you NEED create a manyToMany model.

**Reply by thetutlage** [reply]
Yes, there is no need to create an additional model. Here are the relevant docs on inserting additional data to the pivot table https://docs.adonisjs.com/reference/orm/relations/many-to-many#createmany

**Reply by willonf** [reply]
This is the answer that I was needing! Thank you!
